### DISCUSSION TITLE
RSS feed

### DISCUSSION THREAD

**Discussion by iamgiannisx** [razonyang/hugo-theme-bootstrap#694] [General]
I add my rss feed to a reader (Feedly).

1. The feed list is fetched without images.
2. I can't see the full articles inside feed reader, so my question is, is working like that or is a bug.

**Comment by razonyang** [comment]
Yep, it's working as expected, because the theme uses Hugo built-in [RSS](https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/_default/rss.xml) template for rendering the RSS feed.

As the requirements of customizing RSS comes up, it may be useful that create a custom RSS template inside the theme.

**Comment by razonyang** [comment]
There is a solution #401 to show the full article in RSS feed. 

Custom RSS template #695 is planned, but it won't be implemented quickly.

**Comment by razonyang** [comment]
@iamgiannisx Hi, I just created a PR #697 for this. Please feel free to test it and share your feedback.

`hugo mod get github.com/razonyang/hugo-theme-bootstrap@feat-custom-rss` 

or 

```
$ cd themes/hugo-theme-bootstrap
$ git fetch
$ git checkout feat-custom-rss
```

You may need to enable the following parameter to show full content in feed reader.

```
[feeds]
  contentAsDescription = true
```

@Natenom CC, You may be interested in this.

**Comment by Natenom** [comment]
Thank you for mentioning :)

Tried it and added the feeds part to my config and renamed my `rss.xml` to `_disabled_rss.xml`, but there are excerpts only in the feed.

**Reply by razonyang** [reply]
@Natenom Could you please check the generated `index.xml` source code? I believe there are caches in RSS reader client. Btw, the following parameter is disabled by default.

```
[feeds]
  contentAsDescription = true
```

**Comment by Natenom** [comment]
For testing feeds I use chromium or wget, so it was not cached.

Is the feeds part a subthing of another configuration area? It is the last entry in my config file.

**Reply by razonyang** [reply]
Hmm, that's weird, I could not reproduce it, have you enabled the `feeds.contentAsDescription` parameter and use the right branch `feat-custom-rss`?
I've tested it in local and online, it works as expected.
If you can see full content present inside the `<description>` tag in `http://localhost:1313/index.xml` or `http://localhost:1313/en/index.xml`, then it works(restart `hugo server` and force refresh the page to ensure that was not a cache).

**Comment by Natenom** [comment]
Works. First, tested if your implementation was used at all by adding a string to it. Then I saw, that it is a subthing of params, changed the config from

```
[feeds]
  contentAsDescription = true
```
to
```
feeds.contentAsDescription = true                                                                                                                                                                                                      
```

which is now within the [params] section and it works :)

**Reply by razonyang** [reply]
The first syntax should be correct, I didn't use the second one, I'm not sure that if it works.

**Comment by Natenom** [comment]
What works for me is:

```
[params]
bla = 1
foo = 2
[params.feeds]
contentAsDescription = true
```

This did not work:

```
[params]
bla = 1
foo = 2
[feeds]
contentAsDescription = true
```

**Reply by razonyang** [reply]
I see, you're placing the parameters in `config.toml`. It is correct syntax in `config.toml`. And the following works on `params.toml`:

```
[feeds]
  contentAsDescription = true
```

**Comment by iamgiannisx** [comment]
Hi @razonyang,

Thanks for your response with a solution.

Tested and with standard description works as expected, but if you set content as description the things being a little mess-up, you can check the results with https://codebeautify.org/rssviewer.

You can find a well working xml feed at https://www.smashingmagazine.com/feed in order to check for differences, I checked and full description has content:encoded tag for full content and not description, also has <![CDATA in front of texts.

Thanks.

**Reply by razonyang** [reply]
Thanks for the information, it looks prettier than escaped HTML markup. Done.
I'll merge that PR later if there is no any other issues.

**Comment by razonyang** [comment]
@Natenom @iamgiannisx Thanks for your help and feedback, PR merged, BTW, the `feeds.contentAsDescription` parameter was renamed to `feeds.content`.
